astronomer/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -0,0 +1,157 @@
+    :param env_vars: Environment variables for spark-submit. It
+        supports yarn and k8s mode too.
+    :param verbose: Whether to pass the verbose flag to spark-submit process for debugging
+    :param spark_binary: The command to use for spark submit.
+                         Some distros may use spark2-submit.
+    """
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    async def spark_submit_job(self, application: str, **kwargs: Any) -> None:
+        """
+        Asyncio subprocess create_subprocess_shell to execute the spark-submit job
+
+        :param application: Submitted application, jar or py file
+        :param kwargs: extra arguments to create_subprocess_shell (see subprocess.create_subprocess_shell)
+        """
+        spark_submit_cmd = self._build_spark_submit_command(
+            application
+        )  # inherit from the SparkSubmitHook class
+        # to build the spark submit command
+
+        spark_cmd = " ".join(spark_submit_cmd)
+
+        self._submit_sp = await asyncio.create_subprocess_shell(
+            spark_cmd,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+            bufsize=0,
+            **kwargs,
+        )
+        self.log.debug("Started subprocess: %r", self._submit_sp)
+        await asyncio.wait(
+            [
+                asyncio.create_task(self._process_submit_log(self._submit_sp.stdout)),
+                asyncio.create_task(self._process_submit_log(self._submit_sp.stderr)),
+            ]
+        )
+        return_code = self._submit_sp.returncode
+
+        if return_code:
+            raise AirflowException(
+                f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {return_code}."
+            )
+        if (
+            return_code is None
+            and self._connection["master"] == "local[*]"
+            and self._should_track_driver_status is False
+        ):
+            self._driver_status = "FINISHED"
+
+    async def _process_submit_log(self, stream: Any) -> None:
+        """
+        Processes the log and extracts useful information out of it.
+
+        :param stream: subprocess stream bytes
+        """
+        while True:
+            line = await stream.readline()
+            if line:
+                line = line.decode().strip()
+                line = line.strip()
+                self.log.info(line)
+                # TODO
+                # Need to implement the cluster mode yarn, standalone cluster mode, kubernetes
+                # Need to get the application id if cluster_mode is in yarn mode
+                # Need to get the driver id if cluster_mode is in standalone and kubernetes
+            else:
+                break
+
+    def track_driver_status(self) -> Optional[str]:
+        """Track driver status by driver id"""
+        if self._should_track_driver_status:
+            if self._driver_id is None:
```

Comment:
```suggestion
            if not self._driver_id:
```
Can we use this instead? Python will ensure that it has non-null, non-empty value (in case of string: empty strings, in case of boolen: False and so on)